### PR TITLE
feat: Implement Dialog animations

### PR DESCRIPTION
**Pull request by Fabricevladimir**
## Description

- adds animations for the `Dialog` and `DialogAlert` components

## Type of change

-   [x] New feature (non-breaking change which adds functionality)

## Checklist:

-   [x] I have selected the correct base branch.
-   [x] I have performed a self-review of my own code.
-   [x] I have commented my code, particularly in hard-to-understand areas.
-   [x] I have made corresponding changes to the documentation.
-   [x] My changes generate no new warnings.
-   [x] Any dependent changes have been merged and published in downstream modules.
-   [x] I ran all tests with success and extended them if necessary.
-   [x] I have updated the `CHANGELOG.md` file in the root folder of the package after the [UPCOMING] title and before
        the latest version.
-   [ ] I have tested my code on the test network.

### CODE REVIEW

File: src/core/components/dialogs/dialog/dialogRoot/dialogRoot.tsx [hunk 1]
```diff
@@ -50,6 +51,16 @@ export interface IDialogRootProps extends ComponentPropsWithoutRef<'div'> {
...
 
+const overlayAnimationVariants = {
+    closed: { opacity: 0 },
+    open: { opacity: 1 },
+};
+
+const contentAnimationVariants = {
+    closed: { opacity: 0, scale: 0.88, y: 100 },
+    open: { opacity: 1, scale: 1, y: 0 },
+};
```
Comment (cgero-eth) [inline src/core/components/dialogs/dialog/dialogRoot/dialogRoot.tsx:62]: Should we move these two objects to a `dialogUtils` file under `/dialogs` to reuse the same animation settings for both dialogs? I don't think we'll have different animations for dialogs & alert-dialogs

File: src/core/components/dialogs/dialog/dialogRoot/dialogRoot.tsx [hunk 2]
```diff
@@ -68,23 +79,45 @@ export const DialogRoot: React.FC<IDialogRootProps> = (props) => {
...
-                    onEscapeKeyDown={onEscapeKeyDown}
-                    onInteractOutside={onInteractOutside}
-                    onOpenAutoFocus={onOpenAutoFocus}
-                    onPointerDownOutside={onPointerDownOutside}
-                >
-                    {children}
-                </Content>
-            </Portal>
+            <AnimatePresence>
+                {rootProps.open && (
```
[the remark this answers is not in the document]
Comment (Fabricevladimir) [reply src/core/components/dialogs/dialog/dialogRoot/dialogRoot.tsx:83]: Yes and no. The `AnimatePresence` is needed to for visibility animations, specifically when we want to animate a component when it gets unmounted. In order for the closing animations to work, we have to `forceMount` (which I forgot to do oops!) the `Portal` which means then the `rootProps.open` logic is what actually takes care of opening and closing the dialog.

That said, let me confirm with design whether we actually want the closing animation

[the remark this answers is not in the document]
Comment (Fabricevladimir) [reply src/core/components/dialogs/dialog/dialogRoot/dialogRoot.tsx:74]: Closing animation confirmed to be simply a fade.

**Review by cgero-eth — APPROVED** [review]
LGTM, thanks @Fabricevladimir!

**Pull request merged**